### PR TITLE
Add support for getting balance at a specific date

### PR DESCRIPTION
**Pull request by rvijayc**
I am creating a pull request that adds support to get account balance at a specific date. With this, we can - for example - get income (or expense) for a given year by subtracting the account balance at the start of the year from the account balance at the end of the year.

Please let me know if you have any questions.

### CODE REVIEW

**Comment by sdementen** [comment]
Hello @rvijayc , could you add a corresponding test for the feature?

**Comment by sdementen** [comment]
hello @gregwalters 
thank you for adding tests. however your tests are not testing the functionality you added (the at_date argument and its effect on the results).
can you add tests that specifically test that at_date is taken into account correctly ? on the function itself, can you also add the documentation of the new at_date argument ?

**Comment by dpslwk** [comment]
use 
`post_date=date(`

not
`post_date=datetime(`

in the tests

**Comment by gregwalters** [comment]
Yeah, sorry about the noise. I've been a bit lazy about getting the tests running locally :-\

File: tests/test_account.py [hunk 1]
```diff
@@ -67,9 +68,47 @@ def test_create_standardasset_account(self, new_book):
...
+        racc = book_basic.root_account
+        acc = book_basic.accounts(name="asset")
+        exp = book_basic.accounts(name="exp")
+
+        splits = [
+            Split(account=acc, value=100, memo=u"m�mo asset"),
+            Split(account=exp, value=-100, memo=u"m�mo exp"),
+        ]
+
+        with pytest.raises(GncValidationError):
```
Comment (dpslwk) [inline tests/test_account.py:86]: remove this line and fix indentation of below

File: tests/test_account.py [hunk 2]
```diff
@@ -67,9 +68,47 @@ def test_create_standardasset_account(self, new_book):
...
+            tr = Transaction(currency=EUR, description=u"wire from H�l�ne", notes=u"on St-Eug�ne day",
+                             post_date=datetime(2014, 1, 1),
```
Comment (dpslwk) [inline tests/test_account.py:88]: `post_date=date(2014, 1, 1),`

File: tests/test_account.py [hunk 3]
```diff
@@ -67,9 +68,47 @@ def test_create_standardasset_account(self, new_book):
...
+            tr = Transaction(currency=EUR, description=u"wire from H�l�ne", notes=u"on St-Eug�ne day",
+                             post_date=datetime(2014, 1, 2),
+                             enter_date=datetime(2014, 1, 2),
+                             splits=splits)
+
+        with pytest.raises(GncValidationError):
+            tr = Transaction(currency=EUR, description=u"wire from H�l�ne", notes=u"on St-Eug�ne day",
+                             post_date=datetime(2014, 1, 3),
+                             enter_date=datetime(2014, 1, 3),
+                             splits=splits)
```
Comment (dpslwk) [inline tests/test_account.py:102]: same fix as above

File: tests/test_account.py [hunk 4]
```diff
@@ -1,8 +1,9 @@
 # -*- coding: latin-1 -*-
 import pytest
+from datetime import datetime, date
 
-from piecash import Account, Commodity
-from test_helper import db_sqlite_uri, db_sqlite, new_book, new_book_USD, book_uri
+from piecash import Account, Commodity, Split, Transaction, GncValidationError
```
Comment (dpslwk) [inline tests/test_account.py:5]: remove un-needed `, GncValidationError` once below changes are done

**Review by dpslwk — CHANGES_REQUESTED** [review]

**Comment by gregwalters** [comment]
Yo @sdementen it looks like the CI stuff is a bit hung up. Otherwise this seems ready to merge. Thoughts?

**Comment by sdementen** [comment]
tx you all for your contributions!

**Pull request merged**